lib/blacklight/search_builder.rb

```diff
@@ -27,7 +27,8 @@ def initialize(*options)
       end
 
       @blacklight_params = {}
-      @search_state = Blacklight::SearchState.new(@blacklight_params, @scope&.blacklight_config, @scope)
+      search_state_class = @scope&.search_state_class || Blacklight::SearchState
+      @search_state = search_state_class.new(@blacklight_params, @scope&.blacklight_config, @scope)
```

Comment:
I wonder if it would be cleaner if we gave `Blacklight::SearchService` a `default_search_state`/`empty_search_state`/etc method?

In 8.x, it looks like we could update `with` and `where` to require an existing `SearchState` instance.

Reply:
I think that's a better forward-looking approach; here I'm mostly trying to conform to the existing API (although SearchService and Blacklight::Controller drift a little bit here). I'm totally willing to pivot to `empty_search_state` early, though, if you're 👍  on adding some API in 7.x

Reply:
Another option is to move both `search_state_class` and `search_builder_class` onto `Blacklight::Configuration`, and have the controller and search service both delegate there.

Reply:
Adding a commit here that:
- consolidates the class configurations into a dedicated config
- adds a search state factory method to `Blacklight::Searchable` and `Blacklight::SearchService`
- removes the direct access to configured search state classes in favor of `SearchState#reset`